pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -291,7 +291,7 @@ func (r *ReconcileStorageCluster) ensureCephObjectStoreUsers(instance *ocsv1.Sto
 	if err != nil {
 		return err
 	}
-	if platform == PlatformAWS {
+	if isValidCloudPlatform(platform) {
```

Comment:
This will disable it for UPI environments too where the type will by `PlatformUnknown`. @obnoxxx is that is what is currently recommended? If not, I suggest with going with an approach similar to that in `ensureCephObjectStores`

Reply:
First of all, it is a bit counter-intuitive to have a function `isValidCloudPlatform()` return true for the valid cloud platforms **and for "unknown"** ...

Then, I agree that `ensureCephObjectStores()` and `ensureCephObjectStoreUsers()` should use the same mechansim. Ideally that's a function called in both places. So I'd expect the PR to factor the checking out of `ensureCephObjectStores()` in a first patch and then use that new function in `ensureCephObjectStoreUsers()`. That function could be a "method" under CloudPlatform itself.

Reply:
@crombus , @rohantmp

Reply:
@obnoxxx  how about these changes https://github.com/openshift/ocs-operator/compare/65a83a40a6a924eca1d869187a4a26d0e9a3b499..190d683067652680d61db83ae6a441a87528f476

Reply:
@crombus That is the right direction, but I still sense some code duplication. The whole for loop can go into a function. the slightly confusing bit is that then the function is extremely similar to the existing `isValidCloudPlatform()`, just without the block

```
if p == PlatformUnknown {
    return true
}
```

So, instead of your `GetValidCloudPlatforms`, lets do something like:

```
func isValidCloudPlatformAndNotUnknown(p CloudPlatformType) bool {
    for _, cp := range ValidCloudPlatforms {
        return true
    }
    return false
}
```
And use this in `ensureCephObjectStores` like this:
```
    ...
    platform, err := r.platform.GetPlatform(r.client)
    if err != nil {
        return err
    }
    if isValidCouldPlatformAndNotUnknown(platform) {
        r.reqLogger.Info(fmt.Sprintf("not creating a CephObjectStore because the platform is %T", platform))
        return nil
    }
    ...
```

We just need a better name for the function `isValidCouldPlatformAndNotUnknown`, e.g. rename `isValidCloudPlatform` and use that name ;-)

Reply:
Okay... After all this back and forth, I just realized that when following @rohantmp 's recommendation to do what we do in `ensureCephObjectStores()`, I now realized that I had been looking at old code, and that this is using `isValidCoudPlatform()` since PR #505.... 

So doing what @crombus proposed in this PR makes this completely consistent with what `ensureCephObjectStores()` already does.

We can have a separate discussion of whether we want to include or exclude `unknown`.

Reply:
isValidCloudPlatform was originally just to ensure that the cached value was in the list of types.

We could have is isKnownCloudPlatform